eessi_bot_event_handler.py

```diff
@@ -60,9 +69,137 @@ def handle_issue_comment_event(self, event_info, log_file=None):
-        comment_txt = request_body['comment']['body']
-        self.log("Comment posted in %s by @%s: %s", issue_url, comment_author, comment_txt)
+        action = request_body['action']
+        sender = request_body['sender']['login']
+        owner = request_body['comment']['user']['login']
+        # next two lines are commented out and replaced to write less to log files
+        # txt = request_body['comment']['body']
+        # self.log(f"Comment in {issue_url} (owned by @{owner}) {action} by @{sender}: {txt}")
+        self.log(f"Comment in {issue_url} (owned by @{owner}) {action} by @{sender}")
+
+        # check if addition to comment includes a command for the bot, e.g.,
+        #   bot: build [arch:intel] [instance:AWS]
+        #   bot: cancel [job:jobid]
+        #   bot: disable [arch:generic]
+        # actions: created, edited
+        # created -> comment.body
+        # edited -> comment.body - changes.body.from
+        # procedure:
+        #  - determine what's new (assume new is at the end)
+        #  - scan what's new for commands 'bot: COMMAND [ARGS*]'
+        #  - process commands
+
+        # first check if sender is authorized to send any command
+        # - double purpose:
+        #   1. check permission
+        #   2. skip any comment updates that were done by the bot itself --> we
+        #      prevent the bot entering an endless loop where it reacts on
+        #      updates to comments it made itself
+        #      NOTE this assumes that the sender of the event is corresponding to
+        #      the bot if the bot updates comments itself and that the bot is not
+        #      given permission in the configuration setting 'command_permission'
+        #      ... in order to prevent surprises we should be careful what the bot
+        #      adds to comments, for example, before updating a comment it could
+        #      run the update through the function checking for a bot command.
+        if check_command_permission(sender) is False:
+            self.log(f"account `{sender}` has NO permission to send commands to bot")
+            return
+        else:
+            self.log(f"account `{sender}` has permission to send commands to bot")
+
+        # only scan for comments in newly created comments
+        if action == 'created':
+            comment_new = request_body['comment']['body']
+            # next line commented out and replaced to write less to log files
+            # self.log(f"comment created:\n########\n{comment_new.rstrip()}\n########")
+            self.log(f"comment action '{action}' is handled")
+        else:
+            self.log(f"comment action '{action}' not handled")
+            return
+
+        # search for commands in what is new in comment
+        # init comment_update with an empty string or later split would fail if
+        # it is None
+        comment_update = ''
+        commands = []
+        for line in comment_new.split('\n'):
```

Comment:
Can skip empty lines already here:

```python
        for line in [x for x in [y.strip() in comment_new.split('\n')] if x]:

Reply:
Hmm, this doesn't look complete to me, maybe rather
```python
for line in [x for x in [y.strip() for y in comment_new.split('\n')] if x]:
```
I'll give it a try. Will add a short comment so it will be understandable later too.